java/bulk-export/bulk-export-core/src/main/java/sleeper/bulkexport/model/BulkExportLeafPartitionQuery.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.bulkexport.model;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * An export query for a leaf partition. The query contains information about
+ * which files should be read.
+ * That query is broken down over leaf partitions into subqueries. Each
+ * subquery retrieves records from a leaf partition.
+ */
+public class BulkExportLeafPartitionQuery {
+
+    private final String tableId;
+    private final String exportId;
+    private final String subExportId;
+    private final String leafPartitionId;
+    private final List<String> files;
+
+    private BulkExportLeafPartitionQuery(Builder builder) {
+        tableId = requireNonNull(builder.tableId, builder, "tableId field must be provided");
+        exportId = requireNonNull(builder.exportId, builder, "exportId field must be provided");
+        subExportId = requireNonNull(builder.subExportId, builder, "subExportId field must be provided");
+        leafPartitionId = requireNonNull(builder.leafPartitionId, builder, "leafPartitionId field must be provided");
+        files = requireNonNull(builder.files, builder, "files field must be provided");
+    }
+
+    /**
+     * Checks that the object created is valid and has all of the required fields.
+     *
+     * @return the validated object.
+     */
+    public BulkExportLeafPartitionQuery validate() {
+        return BulkExportLeafPartitionQuery.builder()
+                .tableId(tableId)
+                .exportId(exportId)
+                .subExportId(subExportId)
+                .leafPartitionId(leafPartitionId)
+                .files(files)
+                .build();
+    }
+
```

Comment:
As rest of file already has the javadoc could be relatively simple to add them for these methods and have the whole file covered